### PR TITLE
Addon generator code update

### PR DESCRIPTION
**Pull request by benkeen**
This PR updates the addon generation code:

- now includes a simple “hello world” React component and ties
it in with the route object
- remove legacy doc & code
- gets it up to our new coding standards
- makes it a lot easier to follow + understand

Also updates the writing_addons.md doc page.

### CODE REVIEW

**Comment by benkeen** [comment]
@robertkowalski see what you think of this. The update to the addon generator was long overdue, but the change to the main readme is probably a bit more contentious. It seemed strange to have bits and pieces there so I've left only the key important info about Fauxton: how to install, build, deploy. Everything else is linked to in a section at the end.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -36,10 +41,14 @@ issue
 Closes COUCHDB-XXXX (if there is a Jira ticket)
 ```
 
-When you're ready for a review, submit a Pull Request. We regularly check the PR list for Fauxton and should get back to you with a code review.  If no one has responded to you yet, you can find us on IRC in #couchdb-dev.  Ping **Garren**, **robertkowalski** or **Deathbear**, though anyone in the room should be able to help you.
+When you're ready for a review, submit a Pull Request. We regularly check the PR list for Fauxton and should get back 
+to you with a code review.  If no one has responded to you yet, you can find us on IRC in #couchdb-dev.  
+Ping **Garren** or **robertkowalski** though anyone in the room should be able to help you.
```
Comment (robertkowalski) [inline CONTRIBUTING.md]: michellep :)

Comment (robertkowalski) [inline CONTRIBUTING.md]: i was meaning, michellep is also there

File: tasks/addon/root/routes.js.underscore [hunk 2]
```diff
@@ -11,30 +11,35 @@
...
 
-    crumbs: [
-    ],
+function (app, FauxtonAPI, {%= name %}, Components) {
 
+  var {%= name.substr(0, 1).toUpperCase() + name.substr(1)%}RouteObject = FauxtonAPI.RouteObject.extend({
+    layout: 'one_pane',
+    roles: [],
     routes: {
+      'example/addon': 'helloWorld'
```
Comment (robertkowalski) [inline tasks/addon/root/routes.js.underscore]: i was suprised by the route when testing it...

how about naming a route like: 

`foo: 'helloWorld'` for a module that is named `foo`

```js
routes: {
  '{%= name %}': 'helloWorld'
```

i think?

**Comment by robertkowalski** [comment]
cool stuff, just added an idea

we shoudl try to avoid mixing multiple changes/topic into one commit in the future

**Comment by benkeen** [comment]
^^ Agreed on all points. Reg mixing topics, yeah... I started updating the doc then got carried away & updated the main readme too. I'll actually separate that out into a separate commit.

Also, apologies... but I'm going to make one more addition to this PR then ask you to look at this again. I think I'll bake in adding a primary nav item for the plugin. Just so people can see how that piece works.

File: tasks/addon/root/routes.js.underscore (hunk 2 above)
Comment (benkeen) [inline tasks/addon/root/routes.js.underscore]: I'll make it example/{name} just to prevent any possible conflicts.

**Comment by benkeen** [comment]
Updated.

**Comment by robertkowalski** [comment]
+1

**Comment by benkeen** [comment]
Merged as 78989e6

**Pull request closed without merging**